compiler-rt/lib/scudo/standalone/wrappers_c.inc

```diff
@@ -163,23 +208,11 @@ INTERFACE WEAK void *SCUDO_PREFIX(pvalloc)(size_t size) {
 }
 
 INTERFACE WEAK void *SCUDO_PREFIX(realloc)(void *ptr, size_t size) {
-  if (!ptr) {
```

Comment:
I was hesitating if we want to do that. My argument is that these two cases are defined in the realloc spec so it's still part of the realloc logic.

For some cases (may not be useful through...), you can tell the differences between deallocations from `free()` and `realloc`.

Reply:
BTW, I don't have strong opinion on this. Let me know if you think it's better to exclude them

Reply:
No strong opinion either, but I'd lean on the side of handling them separately. It would make it easier to describe the contract of the new hooks (always `__scudo_realloc_begin_hook` followed by `__scudo_deallocate_hook` followed by `__scudo_allocate_hook` followed by `__scudo_realloc_end_hook`) and client code wouldn't have to second-guess what the intent of the realloc call was.

Reply:
I wound't say there's the contract `always __scudo_realloc_begin_hook followed by __scudo_deallocate_hook followed by __scudo_allocate_hook followed by __scudo_realloc_end_hook` BTW. It's still reasonable to do it in old way but with the scope markers. To me, it says the begin/end for realloc, covering the whole range then it does let the user doesn't need to second-guess what the intent is.

But like I said, no strong opinion here. Will change it to the form desired